Alignment/OfflineValidation/python/TkAlAllInOneTool/DMR_cfg.py

```diff
@@ -0,0 +1,292 @@
+        Nbinx = cms.int32(5000), xmin = cms.double(-0.05), xmax = cms.double(0.05)
+    ),
+    # Normalized Y residuals native coordinates (Pixel)
+    TH1NormYResPixelModules = cms.PSet(
+        Nbinx = cms.int32(120), xmin = cms.double(-3.0), xmax = cms.double(3.0)
+    ),
+    # Y residuals native coordinates (Pixel)
+    TH1YResPixelModules = cms.PSet(
+        Nbinx = cms.int32(5000), xmin = cms.double(-0.05), xmax = cms.double(0.05)
+    ),
+    # X Residuals vs reduced local coordinates (Strip)
+    TProfileXResStripModules = cms.PSet(
+        Nbinx = cms.int32(34), xmin = cms.double(-1.02), xmax = cms.double(1.02)
+    ),
+    # X Residuals vs reduced local coordinates (Strip)
+    TProfileYResStripModules = cms.PSet(
+        Nbinx = cms.int32(34), xmin = cms.double(-1.02), xmax = cms.double(1.02)
+    ),
+    # X Residuals vs reduced local coordinates (Pixel)
+    TProfileXResPixelModules = cms.PSet(
+        Nbinx = cms.int32(17), xmin = cms.double(-1.02), xmax = cms.double(1.02)
+    ),
+    # X Residuals vs reduced local coordinates (Pixel)
+    TProfileYResPixelModules = cms.PSet(
+        Nbinx = cms.int32(17), xmin = cms.double(-1.02), xmax = cms.double(1.02)
+    ),
+)
+
+##Define sequences depending on validation mode
+if valiMode == "StandAlone":
+    ##Output file
+
+    process.TFileService = cms.Service("TFileService",
+            fileName = cms.string("{}/DMR.root".format(config.get("output", os.getcwd()))),
+            closeFileFast = cms.untracked.bool(True),
+    )
+
+    seqTrackerOfflineValidation = cms.Sequence(process.TrackerOfflineValidation)
+
+if valiMode == "DQM":
+    TrackerOfflineValidationSummary = cms.EDAnalyzer("TrackerOfflineValidationSummary",
+        moduleDirectoryInOutput = cms.string("Alignment/Tracker"),
+        useFit = cms.bool(False),
+        stripYDmrs = cms.bool(False),
+        minEntriesPerModuleForDmr = cms.uint32(100),
+   
+        # DMR (distribution of median of residuals per module) of X coordinate (Strip)
+        TH1DmrXprimeStripModules = cms.PSet(
+            Nbinx = cms.int32(50), xmin = cms.double(-0.005), xmax = cms.double(0.005)
+        ),
+   
+        # DMR (distribution of median of residuals per module) of Y coordinate (Strip)
+        TH1DmrYprimeStripModules = cms.PSet(
+            Nbinx = cms.int32(50), xmin = cms.double(-0.005), xmax = cms.double(0.005)
+        ),
+   
+        # DMR (distribution of median of residuals per module) of X coordinate (Pixel)
+        TH1DmrXprimePixelModules = cms.PSet(
+            Nbinx = cms.int32(50), xmin = cms.double(-0.005), xmax = cms.double(0.005)
+        ),
+   
+        # DMR (distribution of median of residuals per module) of Y coordinate (Pixel)
+        TH1DmrYprimePixelModules = cms.PSet(
+            Nbinx = cms.int32(50), xmin = cms.double(-0.005), xmax = cms.double(0.005)
+        )
+    )
+
+    ## DQM file saver
+    from DQMServices.Core.DQM_cfg import *
+    DqmSaverTkAl = cms.EDAnalyzer("DQMFileSaver",
```

Comment:
is this whole block needed?

Reply:
You mean if entire DQM_cfg needs to be imported?

Reply:
No, I mean that I don't understand why you need to have the` DQMFileSaver` here, since we don't run any DQM module.

Reply:
True, `valiMode` is always set to "StandAlone". I will then remove all code related to DQM